controllers/storagecluster/provider_server.go

```diff
@@ -150,6 +154,12 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
 			desiredService.Spec.ClusterIP = actualService.Spec.ClusterIP
 			desiredService.Spec.IPFamilies = actualService.Spec.IPFamilies
 
+			if actualService.Annotations == nil {
+				actualService.Annotations = desiredService.Annotations
+			} else {
+				actualService.Annotations[serviceCertAnnotation] = ocsProviderCertSecretName
+			}
+
```

Comment:
We might want to support more annotation in the template in the future:
```suggestion
			if actualService.Annotations == nil {
				actualService.Annotations = map[string]string{}
			} 
			for key, value := range desiredService.Annotations {
				actualService.Annotations[key] = value
			}
```